### PR TITLE
Webkul buyer onboarding

### PR DESCRIPTION
**Pull request by dhiraj-toptal**
Webkul module related changes for Buyer on-boarding and Balance payment methods display

### CODE REVIEW

File: Controller/Buyer/Qualify.php [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        $customer->updateData($customerData);
+        $customerResource = $this->customerFactory->create();
+        $customerResource->saveAttribute($customer, 'buyer_id');
+        $this->customerSession->setBuyerId($buyerId);
+        $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/test.log');
+        $logger = new \Zend_Log();
+        $logger->addWriter($writer);
+        $logger->info('$buyerId Qualify session');
+        $logger->info($this->customerSession->getBuyerId());
+        return $buyerId;
```
Comment (ynaxonBalance) [inline Controller/Buyer/Qualify.php]: I think you can insert these lines inside the try block

File: Controller/Buyer/Qualify.php [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        }
+        $buyerId = $response['id'] ?? '';
+        $customer = $this->customer->load($this->customerSession->getCustomer()->getId());
+        $customerData = $customer->getDataModel();
+        $customerData->setCustomAttribute('buyer_id', $buyerId);
+        $customer->updateData($customerData);
+        $customerResource = $this->customerFactory->create();
+        $customerResource->saveAttribute($customer, 'buyer_id');
+        $this->customerSession->setBuyerId($buyerId);
+        $writer = new \Zend_Log_Writer_Stream(BP . '/var/log/test.log');
```
Comment (ynaxonBalance) [inline Controller/Buyer/Qualify.php]: just making sure – should `/var/log/test.log` is the correct file path?

File: CustomerData/QualificationSection.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        $buyerResponse = $this->balancepayHelper->getBuyerAmount();
+        if (!empty($buyerResponse['qualificationStatus']) && $buyerResponse['qualificationStatus'] == 'completed'){
+            $status = true;
+            $showButton = false;
+            $showCredit = true;
+            $creditLimit =  $this->balancepayHelper->formattedAmount($buyerResponse['qualification']['creditLimit'] ?? 0);
+        }
+        return [
+            'creditLimit' => 'Terms Limit: '.$creditLimit,
+            'status' => $status, // $buyerResponse['qualificationStatus'] ?? ''
```
Comment (ynaxonBalance) [inline CustomerData/QualificationSection.php]: remove comment

File: Helper/Data.php [hunk 4]
```diff
@@ -113,40 +155,67 @@ public function getSellerIdByProductId($productId = '')
...
     /**
-     * Valid domain
+     * GetBuyerAmount
      *
-     * @param string $domainName
-     * @return bool
+     * @return array
      */
-    private function isValidDomain($domainName): bool
+    public function getBuyerAmount()
```
Comment (ynaxonBalance) [inline Helper/Data.php]: Looks like method name is misleading – it does not get buyer amount, it makes an api call to get the session buyer. Please rename accordingly

File: view/frontend/web/js/section.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                success: function (response) {
+                    if (response && response.hasOwnProperty('qualificationLink') && response.qualificationLink) {
+                        $("#popup-modal").empty();
+                        $("#popup-modal").append('<iframe id="qualify-iframe" class="no-scroll" scrolling="no" width="100%" src="' + response.qualificationLink + '" frameborder="0" allowfullscreen=""></iframe>');
+                        $('#popup-modal').modal(options).modal('openModal');
+                    } else {
+                        setTimeout(function() {
+                            customerData.set('messages', {
+                                messages: [{
+                                    text: 'There was a problem generating a qualification link. Please contact the administrator.',
```
Comment (ynaxonBalance) [inline view/frontend/web/js/section.js]: change to: 
`There was a problem starting qualification process. Please try again later or contact us.`

File: view/frontend/web/js/section.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+                                    type: 'error'
+                                }]
+                            });
+                        },1000);
+                    }
+                },
+                error: function (xhr, status, errorThrown) {
+                    customerData.set('messages', {
+                        messages: [{
+                            text: 'There was a problem generating a qualification link. Please contact the administrator.',
```
Comment (ynaxonBalance) [inline view/frontend/web/js/section.js]: change to:
`There was a problem starting qualification process. Please try again later or contact us.
`

**Comment by dhiraj-toptal** [comment]
Hi @ynaxonBalance 
I have updated code with fixes. Please review.

**Review by ynaxonBalance — APPROVED** [review]

**Pull request merged**